test/python/test_operation.py

```diff
@@ -65,6 +65,17 @@ def prepare_inputs_targets_for_rnn_test():
     targets = [t0, t1, t2]
     return inputs, targets, h0
 
+def numpy_unary_ops_backward(func, x, dy, h=0.0005):
```

Comment:
This is not correct.
You can either use this [one](http://cs231n.github.io/optimization-1/#gradcompute), or compute the gradient explicitly, e.g., gradient of cos() is sin()

Reply:
OK, I will change the code using one of your two methods (the link provided or compute the gradient explicitly).

The code I did was only the diagonal of the gradient matrix, which is only suitable for unary operator where one output is determined by one input, without the coupling from other inputs.

Reply:
Changed the code by computing the gradient explicitly, the accuracy check is up to 5 decimals.